docs/source/docker/advanced/datasource_predictions.rst

```diff
@@ -0,0 +1,339 @@
+        + some_directory_containing_irrelevant_things/
+
+
+we can specify which subfolders contain relevant predictions in the `tasks.json`:
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/tasks.json
+
+    [
+        "classification_weather",
+        "classification_scenery",
+        "object_detection_people"
+    ]
+
+.. note::
+
+    If you list a subfolder which doesn't contain a valid `schema.json` file,
+    the Lightly docker will fail! See below how to create a good `schema.json` file.
+
+
+Prediction Schema
+-----------------
+For Lightly it's required to store a prediction schema. The schema helps the Lightly
+Platform to correctly identify and display classes. It also helps to prevent errors
+as all predictions which are loaded are validated against this schema.
+
+
+For classification and object detection the prediction schema must include all the categories and ids.
+For other tasks such as keypoint detection it can be useful to store additional information
+like edges between keypoints.
+
+You can provide all this information to Lightly by adding a `schema.json` to the directory of the respective task.
+
+The schema.json file must have a key categories with a corresponding list of categories following the COCO annotation format.
+For example, let's say we are working with a classification model predicting the weather on an image.
+The three classes are sunny, clouded, and rainy.
+
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/classification_weather/schema.json
+
+    {
+        "categories": [
+            {
+                "id": 0,
+                "name": "sunny",
+            },
+            {
+                "id": 1,
+                "name": "clouded",
+            },
+            {
+                "id": 2,
+                "name": "rainy",
+            }
+        ]
+    }
+
+
+
+Prediction Files
+----------------
+Lightly requires a **single prediction file per image**. The file should be a .json
+following the format defined under :ref:`Prediction Format` and stored in the subdirectory
+`.lightly/predictions/${TASK_NAME}` in the storage bucket the dataset was configured with.
+In order to make sure Lightly can match the predictions to the correct source image,
+it's necessary to follow the naming convention:
+
+.. code-block:: bash
+
+    # filename of the prediction for image FILENAME.EXT
+    .lightly/predictions/${TASK_NAME}/${FILENAME}.json
+
+    # example: my_image.png, classification
+    .lightly/predictions/my_classification_task/my_image.json
+
+    # example: my_subdir/my_image.png, classification
+    .lightly/predictions/my_classification_task/my_subdir/my_image.json
+
+
+Prediction Files for Videos
+---------------------------
+When working with videos, Lightly requires a prediction file per frame. Lightly
+uses a naming convention to identify frames: The filename of a frame consists of
+the video filename, the video format, and the frame number (padded with 8 zeros)
```

Comment:
The padding is not fixed but is dynamically determined by the number of frames in the video: `padding = len(str(num_frames))`

Reply:
Fixed by #762